File: src/Punto/CartesianPoint2D.java
package Punto;

public class CartesianPoint2D implements Point2D {
    private double x;
    private double y;

    public CartesianPoint2D(double x, double y){
        this.x = x;
        this.y = y;
    }

    public CartesianPoint2D(Point2D p){
        this.x = p.getX();
        this.y = p.getY();
    }

    public double getX(){
        return x;
    }

    public double getY(){
        return y;
    }

    @Override
    public Point2D somma(Point2D p){
        return new CartesianPoint2D(getX()+p.getX() , getY() + p.getY());
    }

    @Override
    public Point2D copy(){
        return new CartesianPoint2D(getX(), getY());
    }

}

File: Python_Esercizi/Punto/CartesianPoint2D.py
from Punto.AbstractPoint2D import AbstractPoint2D

class CartesianPoint2D(AbstractPoint2D):
    def __init__(self, x, y):
        self._x = x
        self._y = y

    def getX(self):
        return self._x

    def getY(self):
        return self._y

